addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/CacheableScanRule.java

```diff
@@ -697,6 +692,15 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         }
     }
 
+    private Long extractAgeValue(String directiveToken, int tokenLength) {
+        return Long.parseLong(
+                directiveToken.substring(
+                        tokenLength,
+                        directiveToken.indexOf(",", tokenLength) == -1
```

Comment:
- Worth extracting a local variable.
- I thought more would be extracted, e.g. the endsWith and the earlier splits. But fine as is, we should add something to `commonlib` to parse directives and update throughout the codebase.

Reply:
You mean turn the substring into a local var and then return it Long parsed?

(turn this into two steps)

Reply:
Or just the index of the comma so it isn’t computed multiple times?

Reply:
I was referring to `directiveToken.indexOf(",", tokenLength)`, which is called twice (in the worst case).

Reply:
Yeah, missed your latest comment…